pkg/svc/zserver.go

```diff
@@ -2,51 +2,206 @@ package svc
 
 import (
 	"context"
+	"sync"
+	"time"
 
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/my-app/pkg/pb"
+	rpb "github.com/my-app/pkg/resp_pb"
+	"github.com/spf13/viper"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~ A BRIEF DEVELOPMENT GUIDE ~~~~~~~~~~~~~~~~~~~~~~~~~
-//
-// TODO: Extend the MyApp service by defining new RPCs and
-// and message types in the pb/service.proto file. These RPCs and messages
-// compose the API for your service. After modifying the proto schema in
-// pb/service.proto, call "make protobuf" to regenerate the protobuf files.
-//
-// TODO: Create an implementation of the MyApp server
-// interface. This interface is generated by the protobuf compiler and exists
-// inside the pb/service.pb.go file. The "server" struct already provides an
-// implementation of MyApp server interface service, but only
-// for the GetVersion function. You will need to implement any new RPCs you
-// add to your protobuf schema.
-//
-// TODO: Update the GetVersion function when newer versions of your service
-// become available. Feel free to change GetVersion to better-suit how your
-// versioning system, or get rid of it entirely. GetVersion helps make up a
-// simple "starter" example that allows an end-to-end example. It is not
-// required.
-//
-// TODO: Oh yeah, delete this guide when you no longer need it.
-//
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~ FAREWELL AND GOOD LUCK ~~~~~~~~~~~~~~~~~~~~~~~~~~
-// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
 const (
 	// version is the current version of the service
 	version = "0.0.1"
 )
 
 // Default implementation of the MyApp server interface
-type server struct{}
+type server struct {
+	pb.UnimplementedMyAppServer
+	Description string
+	Timestamp   time.Time
+	Requests    int64
+	mtx         sync.RWMutex
+	mtxDescr    sync.RWMutex
```

Comment:
Why do you use 2 different mutexes? For better performance?

Reply:
Yes, i assume its better to create new structers with mutexes, instead of just use mutexes in general structure. Will fix it